TDL-09-shell-EC2/deploy_instance.sh

```diff
@@ -0,0 +1,229 @@
+    ;;
+  centos6)
+    image_id=$(aws ec2 describe-images --owners 679593333241 \
+      --filters \
+        Name=name,Values='CentOS Linux 6 x86_64 HVM EBS*' \
+        Name=architecture,Values=x86_64 \
+        Name=root-device-type,Values=ebs \
+      --query 'sort_by(Images, &Name)[-1].ImageId' \
+      --output text)
+    ;;
+  centos7)
+    image_id=$(aws ec2 describe-images --owners aws-marketplace \
+      --filters \
+        Name=name,Values='CentOS Linux 7 x86_64 HVM*' \
+        Name=product-code,Values=aw0evgkw8e5c1q413zgy5pjce \
+        Name=architecture,Values=x86_64 \
+        Name=root-device-type,Values=ebs \
+      --query 'sort_by(Images, &CreationDate)[-1].[ImageId]' \
+      --output text)
+    ;;
+  ubuntu)
+    image_id=$(aws ec2 describe-images \
+      --owners 099720109477 \
+      --filters Name=name,Values=ubuntu/images/hvm-ssd/ubuntu-xenial-16.04-amd64-server-* \
+      --query 'sort_by(Images,&CreationDate)[-1].ImageId' --output text)
+    ;;
+esac
+echo "Find the latest image of $ostype: ${image_id}"
+ 
+#  Create EC2 instance for osype with instance_type with below settings:
+print_color "green" "-------------------------Create EC2 instance---------------------------"
+# set variables
+KEY_NAME='da-key'
+VPC_NAME='da-c02-vpc'
+VPC_ID=$(aws ec2 describe-vpcs \
+    --filter "Name=tag:Name,Values=$VPC_NAME" \
+    --query Vpcs[].VpcId --output text)
+echo $VPC_ID
+PRIVATE_SUBNET_A_NAME='da-c02-private-a'
+PRIVATE_SUBNET_A_ID=$(aws ec2 describe-subnets  \
+    --filter "Name=tag:Name,Values=$PRIVATE_SUBNET_A_NAME" \
+    --query 'Subnets[].SubnetId' --output text)
+echo "${PRIVATE_SUBNET_A_ID}"
+
+SG_PRIVATE_NAME='da-c02-private-sg'
+SG_PRIVATE_ID=$(aws ec2 describe-security-groups \
+    --filters Name=vpc-id,Values="${VPC_ID}" Name=group-name,Values="${SG_PRIVATE_NAME}" \
+    --query  'SecurityGroups[*].[GroupId]'  --output text)
+echo "${SG_PRIVATE_ID}"
+
+# start_time=$(date -d "$(TZ='Australia/Melbourne' date)" +%Y%m%d%H%M%S)
+# echo ${start_time}
+## AgedTime: yyyymmddhhmm (current time + agedtime days in Australia/Melbourne Time)
+stop_time=$(date -d "$(TZ=$DEFAULT_TIME_ZONE date) $agedtime days -3600 seconds" +%Y%m%d%H%M%S)
+echo ${stop_time}
+
+# Create user_data file
+cat << EOF > ./instance_userdata.txt
+#!/bin/bash
+cat <<EOT >> /etc/motd
+'OSType: $ostype'
+"Please be notified that your instance will be terminated by  ${stop_time}"
+EOT
+EOF
+
+EC2_NAME='ostype_test'
+RULE='ec2-startstop;1900;;all'
+# Create instance
+instance_id=$(aws ec2 run-instances \
+    --image-id "$image_id" \
+    --instance-type "$instance_type" \
+    --count 1 \
+    --subnet-id "$PRIVATE_SUBNET_A_ID" \
+    --key-name  "$KEY_NAME" \
+    --security-group-ids "$SG_PRIVATE_ID" \
+    --no-associate-public-ip-address \
+    --user-data file://instance_userdata.txt  \
+    --tag-specifications "ResourceType=instance,Tags=[{Key=Name,Value=$EC2_NAME},{Key=AgedTime,Value=$stop_time},{Key=scheduler,Value=$RULE}]"  \
```

Comment:
{Key="scheduler:ec2-startstop",Value=$RULE}

Reply:
I have corrected it.